main.go

```diff
@@ -0,0 +1,96 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"strconv"
+
+	"github.com/dispatchframework/funky/pkg/funky"
+)
+
+const (
+	serversEnvVar   = "SERVERS"
+	serverCmdEnvVar = "SERVER_CMD"
+)
+
+type funkyHandler struct {
+	router funky.Router
+}
+
+func (f funkyHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	reader := r.Body
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(reader)
+	rawBody := buf.Bytes()
+	var body map[string]interface{}
+	err := json.Unmarshal(rawBody, &body)
```

Comment:
Could probably use `json.NewDecoder(r.Body).Decode(&body)` instead, so no need to read into another buffer.

Reply:
That is very convenient. I will make this change in a couple other places as well.